src/core/MOM_continuity.F90

```diff
@@ -72,7 +72,7 @@ module MOM_continuity
 
 contains
 
-subroutine continuity(u, v, hin, h, uh, vh, dt, G, CS, uhbt, vhbt, OBC, &
+subroutine continuity(u, v, hin, h, uh, vh, dt, G, CS, diag_cs, uhbt, vhbt, OBC, &
```

Comment:
Could you move the call to diag_update_target_grids() to just after the call to continuity() so that we don't need the diag_CS type passsed into continuity().

This should give just a few less calls to diag_update_target_grids() but more importantly keeps continuity look more like a computational kernel without diagnostic.